go/private/rules/nogo.bzl

```diff
@@ -76,7 +74,7 @@ def _nogo_impl(ctx):
     nogo_source = go.library_to_source(go, struct(
         srcs = [struct(files = [nogo_main])],
         embed = [ctx.attr._nogo_srcs],
-        deps = analyzer_archives,
+        deps = ctx.attr.deps,
```

Comment:
Doesn't this have a different type (list of `Target`s instead of list of providers)? How does this work?

Reply:
The code can handle both; I think it's nicer to pass the `Target`s and then grab the providers as needed; code like `srcs = [struct(files = [nogo_main])],` seems a bit roundabout. Is there a performance reason not to do that?

Reply:
I don't know whether there is a memory cost to retaining targets here. Probably not as Bazel retains them elsewhere. But it's definitely unusual and a bit of a code smell as it allows downstream consumers to read arbitrary (possibly even non-Go providers). It also won't play with Starlark types. 

If it doesn't matter much, let's keep the single provider in this field.

Reply:
that's fair; the part that I don't like is that the code ends up handling both `Target` and provider list. It seems it will need to change either way to support typing, though maybe it should stop supporting Target and push the extraction into all the callers. I'll revert this part for now; we'll need to change it anyway if we refactor the provider setup.